### PR TITLE
Add theme header comparison check to compatibility logic

### PR DESCRIPTION
**Pull request by mattyrob**
This PR adds a compatibility check to Theme Headers to check the value is not 5.0 or above.

It may need more work to assess what happens if that header is not present.

### CODE REVIEW

File: lib/admin-page.php [hunk 1]
```diff
@@ -453,7 +453,8 @@ function classicpress_check_can_migrate() {
 	$theme = wp_get_theme();
 	if ( isset( $parameters['themes'] ) &&
 		in_array( $theme->stylesheet, (array) $parameters['themes'] ) ||
-		( is_child_theme() && in_array( $theme->parent()->stylesheet, (array) $parameters['themes'] ) )
+		( is_child_theme() && in_array( $theme->parent()->stylesheet, (array) $parameters['themes'] ) ) ||
+		version_compare( $theme->get( 'RequiresWP' ), '5.0' ) >= 0
```
Comment (nylen) [inline lib/admin-page.php:457]: As written, this reads the same as when we have added a theme to our hardcoded list of incompatible themes in the API. I think this should be a separate check instead. This check could then indicate which of the possible situations is present:

- The theme actually declares compatibility with WP 4.9+ or earlier (all good)
- The theme declares compatibility with WP 5.0+ or later (bad)
- The theme does not declare a compatibility header (unknown)

This way we can tell the user exactly what is being checked.

Adding more checks will mean that the migration plugin output is a bit long... I actually think this is fine. Migration is a complicated process and there is a lot that can go wrong.

Comment (mattyrob) [reply lib/admin-page.php:457]: So, are you proposing a new section be added to the check steps that outputs an analysis of the theme header and then the next section compares the theme to out known list of incompatibilities as per the API?

Or should we just make the output under the 'theme' heading a little more specific in terms fo telling the user if their theme is declaring a problem or if we are aware of a problem?

**Comment by mattyrob** [comment]
I've done a bit more playing around and extended the logic in the Theme checking.

1/ Display a blocking message if we know there is an issue via the API
2/ Display a warning message if the 'RequiresWP` header is not present (interestingly is is not consistently use in past core themes so we may need to recommend a specific theme - or perhaps better yet, our own theme)
3/ Display a blocking message if the 'RequiresWP` header indicates a higher version (5.0 or above) is needed.
4/ If all the above are not displayed show a message saying we are not aware of any issues.

Updated patch to follow, happy for feedback and further adjustments can be made yet.

File: lib/admin-page.php [hunk 2]
```diff
@@ -459,14 +459,35 @@ function classicpress_check_can_migrate() {
...
+			/* translators: active theme name */
+			'It looks like you are using the <strong>%1$s</strong> theme. We cannot be sure it is compatible with ClassicPress because it is not declaring a minimum required version of WordPress.',
+			'switch-to-classicpress'
+		), $theme->name );
+	} elseif ( version_compare( $theme->get( 'RequiresWP' ), '5.0' ) >= 0 ) {
+		$preflight_checks['theme'] = false;
+		echo "<tr>\n<td>$icon_preflight_fail</td>\n<td>\n";
+		printf( __(
+			/* translators: active theme name */
+			'It looks like you are using the <strong>%1$s</strong> theme. Unfortunately it seems it requires WordPress %2$s or above and it is therefore incompatible with ClassicPress.',
```
Comment (striebwj) [inline lib/admin-page.php:483]:
```suggestion
			'It looks like you are using the <strong>%1$s</strong> theme. Unfortunately it seems it requires WordPress %2$s or above and it therefore may be incompatible with ClassicPress.',
```

Thoughts on leaving a little wiggle room with the wording?

**Review by striebwj — APPROVED** [review]

**Pull request merged**